webapp-django/crashstats/api/models.py

```diff
@@ -1 +1,88 @@
-# deliberately empty so Django thinks this is an app
+import datetime
+
+from django import http
+
+from crashstats.api import forms
+from crashstats.crashstats import models
+from crashstats.supersearch.models import SuperSearch
+
+
+class NewSignatures(models.SocorroMiddleware):
+
+    API_WHITELIST = None
+
+    possible_params = (
+        'start_date',
+        'end_date',
+        'not_after',
+    )
+
+    def get(self, **kwargs):
+        form = forms.NewSignaturesForm(kwargs)
+
+        if not form.is_valid():
+            return http.JsonResponse({
+                'errors': form.errors
+            }, status=400)
+
+        start_date = form.cleaned_data['start_date']
+        end_date = form.cleaned_data['end_date']
+        not_after = form.cleaned_data['not_after']
+
+        if not end_date:
+            end_date = (
+                datetime.datetime.utcnow().date() + datetime.timedelta(days=1)
+            )
+
+        if not start_date:
+            start_date = end_date - datetime.timedelta(days=8)
+
+        if not not_after:
+            not_after = start_date - datetime.timedelta(days=14)
+
+        api = SuperSearch()
+
+        signatures_number = 100
+
+        # First let's get a list of the top signatures that appeared during
+        # the period we are interested in.
+        params = {
+            'product': 'Firefox',
+            'date': [
+                '>=' + start_date.isoformat(),
+                '<' + end_date.isoformat(),
+            ],
+            '_facets': 'signature',
+            '_facets_size': signatures_number,
+            '_results_number': 0,
+        }
+        data = api.get(**params)
+
+        signatures = []
+        for signature in data['facets']['signature']:
+            signatures.append(signature['term'])
+
+        # Now we want to verify whether those signatures appeared or not during
+        # some previous period of time.
+        params['date'] = [
+            '>=' + not_after.isoformat(),
+            '<' + start_date.isoformat(),
+        ]
+
+        # Filter exactly the signatures that we have.
+        params['signature'] = ['=' + x for x in signatures]
```

Comment:
Why should it? I think ``contains`` is a better candidate, and in the SuperSearch world ``has term`` is the *de facto* default for most fields.